### PR TITLE
Coverage support

### PR DESCRIPTION
**Pull request by VeckoTheGecko**
Added coverage support with Codecov.

**Configuration @erikvansebille :**
- [x] Go to [codecov.io](https://about.codecov.io/) and sign up with GitHub
- [x] Setup instructions visible by finding the `OceanParcels/parcels` repo in the UI and clicking configure (you should be at `https://app.codecov.io/gh/OceanParcels/parcels/new`)
  - [x] Step 1. Add [CodeCov GitHub app](https://github.com/apps/codecov) and grant access to repo
  - [x] Step 2. add the Codecov uploader to your CI workflow

### CODE REVIEW

**Comment by VeckoTheGecko** [comment]
Waiting on outcome of #1283, and for successful coverage upload

**Comment by VeckoTheGecko** [comment]
Coverage has been [successfully uploaded](https://app.codecov.io/gh/VeckoTheGecko/parcels) for my fork

**Comment by erikvansebille** [comment]
I think I now did all the steps above, but there's nothing to report yet at https://app.codecov.io/gh/OceanPARCELS/parcels. There is something at https://app.codecov.io/gh/OceanPARCELS/parcels/pulls, though. Anything else I need to do from my side?

**Comment by VeckoTheGecko** [comment]
I think we're good to go. Once merged into master (+triggering CI) it should populate and show the coverage.

Coverage will only show on `master`, `test-me/*` and pulls due to the triggering of the workflows

File: codecov.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    project:
+      default:
+        informational: true
+    patch:
+      default:
+        informational: true
+comment: false # Only on PR when theres a change in coverage
+
+# When modifying this file, please validate using
+# curl -X POST --data-binary @codecov.yml https://codecov.io/validate
```
Comment (erikvansebille) [inline codecov.yml:12]: This needs a newline at the end of file, I think?

```suggestion
# curl -X POST --data-binary @codecov.yml https://codecov.io/validate

```

**Comment by erikvansebille** [comment]
Is this one ready for merging now?

**Comment by VeckoTheGecko** [comment]
Good to go

**Pull request merged**